### PR TITLE
Initial implementation of Spotifiy API connection, including a test suite

### LINKED ISSUE #9: Set-up Spotify API integration
**Is your feature request related to a problem? Please describe.**

We do not currently have a way of connecting to the Spotify API to create playlists, get data, etc.
**Describe the solution you'd like**

We should be able to connect to Spotify's API 
**Additional context**

If an API token or key is required, we should set this up as a secret if possible (do not commit api keys in code)

### PR DESCRIPTION
**Pull request by MattGrantUMich**
This includes a JS based connection between our application and the Spotify API, allowing for easy development upon this structure.

Closes #9 

After your review, we should meet to see the next steps for this project.

### CODE REVIEW

File: src/js/GoldifyExecutePage.js [hunk 1]
```diff
@@ -1,14 +1,196 @@
...
+      client_id: clientId,
+      scope: "user-read-private user-read-email",
+      redirect_uri: redirectUri,
+      state: generateRandomString(16)
+    });
+  replaceWindowURL(spotifyApiURL);
+}
+
+function replaceWindowURL(url) {
+  if (!goldifyUtils.jestTestIsRunning()) {
```
Comment (kyrod) [inline src/js/GoldifyExecutePage.js:45]: I feel weird about adding logic only related to if tests are running - this is fine for an initial go but i'd really like if we can get rid of this somehow, and maybe mock certain functions if we need in the tests themselves

File: src/js/GoldifyExecutePage.js [hunk 2]
```diff
@@ -1,14 +1,196 @@
...
+          replaceWindowURL("/");
+        }
+        return response.json();
+      })
+      .then((data) => {
+        this.retrieveUserData(code, data);
+      })
+      .catch(error => console.log(error)); // eslint-disable-line no-console
+    if (goldifyUtils.jestTestIsRunning()) {
+      this.setStateForTest();
```
Comment (kyrod) [inline src/js/GoldifyExecutePage.js:101]: Same here, if we could just have lots of unit tests and mocked function calls instead, we should be able to remove all logic only related to testing

**Review by kyrod — APPROVED** [review]
looks good, we can rework the tests in a future issue

File: src/js/GoldifyExecutePage.js (hunk 1 above)
Comment (MattGrantUMich) [reply src/js/GoldifyExecutePage.js]: This is a great point, and not the right move by me. Could get very sloppy.

For record keeping, this has been resolved by using mocking in our test suites.

File: src/js/GoldifyExecutePage.js (hunk 2 above)
Comment (MattGrantUMich) [reply src/js/GoldifyExecutePage.js]: Same as comment above

```
This is a great point, and not the right move by me. Could get very sloppy.

For record keeping, this has been resolved by using mocking in our test suites.
```

**Comment by MattGrantUMich** [comment]
Note: I made a small change to our CI workflow config file to include our Github Secrets as environment vars. Turns out it works! I wasn't sure if it was going to based on [https://github.com/actions/starter-workflows/issues/68](this issue thread).

Anyway, you may want to check that too to make sure it looks good. My apologies for editing it directly in master, that won't happen again.

**Comment by MattGrantUMich** [comment]
For future reference:

- Original Review was given approval as we were going to re-factor testing in the future. However, we came to the conclusion that we should get the testing situated now before we keep moving forward. Build a strong foundation.

I have since requested another review from @kyrod, who has much more experience with testing, as well as CI configuration (which I made a note about some changes above).

Kyle, if you see any changes that should be made, I say we get those in before we push through this PR. That way we're pretty concrete with our testing structure before reaaaally diving in. Thanks 💯

**Review by kyrod — APPROVED** [review]
looks good to me bruther

**Comment by MattGrantUMich** [comment]
Closes #10

**Pull request merged**